cpp/vm/evmzero/interpreter_test.cc

```diff
@@ -2845,12 +2848,29 @@ TEST(InterpreterTest, BLOCKHASH) {
       .state_after = RunState::kDone,
       .gas_before = 40,
       .gas_after = 20,
-      .stack_before = {21},
+      .stack_before = {900},
       .stack_after = {0x0a0b0c0d},
       .host = &host,
   });
 }
 
+TEST(InterpreterTest, BLOCKHASH_outOfRange) {
+  MockHost host;
+  EXPECT_CALL(host, get_tx_context())  //
+      .Times(1)
+      .WillOnce(Return(evmc_tx_context{.block_number = 1000}));
+
+  RunInterpreterTest({
+      .code = {op::BLOCKHASH},
+      .state_after = RunState::kDone,
+      .gas_before = 40,
+      .gas_after = 20,
+      .stack_before = {500},
+      .stack_after = {0},
+      .host = &host,
+  });
+}
+
```

Comment:
Please also add test cases for edge cases where the provided number is
- equal to the current block - {0, 1, 255, 256, 257}
- an overflow that would result in a valid value if truncated